src/zjs_util.c

```diff
@@ -77,14 +77,30 @@ void zjs_obj_add_object(jerry_value_t parent, jerry_value_t child,
 void zjs_obj_add_string(jerry_value_t obj, const char *str, const char *name)
 {
     // requires: obj is an existing JS object
-    //  effects: creates a new field in parent named name, set to sval
+    //  effects: creates a new field in parent named name, set to str
     jerry_value_t jname = jerry_create_string((const jerry_char_t *)name);
     jerry_value_t jstr = jerry_create_string((const jerry_char_t *)str);
     jerry_set_property(obj, jname, jstr);
     jerry_release_value(jname);
     jerry_release_value(jstr);
 }
 
+void zjs_obj_add_readonly_string(jerry_value_t obj, const char *str,
+                                 const char *name)
+{
+    // requires: obj is an existing JS object
+    //  effects: creates a new readonly field in parent named name, set to str
+    jerry_value_t jname = jerry_create_string((const jerry_char_t *)name);
+    jerry_property_descriptor_t pd;
+    jerry_init_property_descriptor_fields(&pd);
+    pd.is_writable = false;
+    pd.is_value_defined = true;
+    pd.value = jerry_create_string((const jerry_char_t *)str);
```

Comment:
you don't need to call jerry_release_value() on pd.value? or the jerry_free_property_descriptor_fields() does it?